pygmt/alias.py

```diff
@@ -0,0 +1,134 @@
+    original value will be returned. If ``mapping`` is set to ``True``, the first letter
+    of the long-form argument will be used as the short-form argument.
+
+    An optional prefix (e.g., `"+o"`) can be added to the beginning of the converted
+    string.
+
+    To avoid extra overhead, this function does not validate parameter combinations. For
+    example, if ``value`` is a sequence but ``separator`` is not specified, the function
+    will return a sequence of strings. In this case, ``prefix`` has no effect, but the
+    function does not check for such inconsistencies. The maintaner should ensure that
+    the parameter combinations are valid.
+
+    Parameters
+    ----------
+    value
+        The value to convert.
+    prefix
+        The string to add as a prefix to the returned value.
+    mapping
+        A mapping dictionary or ``True`` to map long-form arguments to GMT's short-form
+        arguments. If ``True``, will use the first letter of the long-form arguments.
+    separator
+        The separator to use if the value is a sequence.
+
+    Returns
+    -------
+    ret
+        The converted value.
+
+    Examples
+    --------
+    >>> _to_string("text")
+    'text'
+    >>> _to_string(12)
+    '12'
+    >>> _to_string(True)
+    ''
+    >>> _to_string(False)
+    >>> _to_string(None)
+
+    >>> _to_string("text", prefix="+a")
+    '+atext'
+    >>> _to_string(12, prefix="+a")
+    '+a12'
+    >>> _to_string(True, prefix="+a")
+    '+a'
+    >>> _to_string(False, prefix="+a")
+    >>> _to_string(None, prefix="+a")
+
+    >>> _to_string("high", mapping=True)
+    'h'
+    >>> _to_string("mean", mapping={"mean": "a", "mad": "d", "full": "g"})
+    'a'
+    >>> _to_string("invalid", mapping={"mean": "a", "mad": "d", "full": "g"})
+    Traceback (most recent call last):
+    ...
+    pygmt...GMTInvalidInput: Invalid value: 'invalid'. Valid values are: mean, ...
+
+    >>> _to_string((12, 34), separator="/")
+    '12/34'
+    >>> _to_string(("12p", "34p"), separator=",")
+    '12p,34p'
+    >>> _to_string(("12p", "34p"), prefix="+o", separator="/")
+    '+o12p/34p'
+
+    >>> _to_string(["xaf", "yaf", "WSen"])
+    ['xaf', 'yaf', 'WSen']
+    """
+    # None and False are converted to None.
+    if value is None or value is False:
+        return None
+    # True is converted to an empty string with the optional prefix.
+    if value is True:
+        return f"{prefix}"
+    # Any non-sequence value is converted to a string.
+    if not is_nonstr_iter(value):
+        match mapping:
+            case False:
+                pass
+            case True:
+                value = value[0]
+            case Mapping():
+                if value not in mapping and value not in mapping.values():
+                    raise GMTValueError(
+                        value,
+                        description="value for parameter {name!r}" if name else "value",
+                        choices=mapping.keys(),
+                    )
```

Comment:
`mapping=True` means using the first letter of a long-form argument as its short-form argument. For example, `resolution="high"` will be mapped to `resolution="h"`. If we don't support `mapping=True`, lines 112-123 can be simplified to a few lines:
```python
                if value not in mapping and value not in mapping.values():
                    raise GMTValueError(
                        value,
                        description="value for parameter {name!r}" if name else "value",
                        choices=mapping.keys(),
                    )
```
but without `mapping=True`, we need to write something like
```
mapping={"full": "f", "high": "h", "intermediate": "i", "low": "l", "crude": "c"}
```
when calling the `_to_string` function.

Reply:
I've removed `mapping=True` in commit c499aab2c1274b1c9b7251439548aa0028c68495.